### PR TITLE
Response examples should respect media-type

### LINKED ISSUE #6442: Response examples do not respect media-type
### Q&A (please complete the following information)
OS: [e.g. macOS] Windows 10 Pro
Browser: [e.g. chrome, safari] chrome
Version: [e.g. 22] Version 85.0.4183.102 (Official Build) (64-bit)
Method of installation: [e.g. npm, dist assets] raw code
Swagger-UI version: [e.g. 3.10.0] master
Swagger/OpenAPI version: [e.g. Swagger 2.0, OpenAPI 3.0] 3.0.1
### Content & configuration

Example Swagger/OpenAPI definition:
```yaml
openapi: 3.0.1
info:
  title: Example Swagger
  version: '1.0'
servers:
  - url: /api/v1
paths:
  /xmlTest:
    get:
      summary: subscribes to a siri vm stream
      operationId: registerSubscription
      parameters: []
      responses:
        '200':
          description: Simple example
          content:
            application/xml:
              schema:
                type: object
                xml:
                  name: root
                properties:
                  x:
                    type: string
                example: 
                  x: what the f
              examples:
                x2:
                  summary: "xml not rendered via 'examples' keyword"
                  value:
                    x: should be xml
```

### Describe the bug you're encountering

When providing examples via `examples` or `example` keyword for responses it will not be rendered according to media-type.

### To reproduce...
Open yaml at editor and try to view xml response sample. You can delete the `examples` part and it will render it in xml.

### Expected behavior
Examples provided via `examples` definition should be rendered respecting the content-type.

### Screenshots

With examples:
image
Without examples, ok:
image

### PR DESCRIPTION
**Pull request by mathis-m**
### Description

Response examples do not respect the media type selected.
It just stringifies the json representation of the sample provided via `examples` or `example` keyword.
It should use the `getSampleSchema` method, which does take care of the content-type // media-type.

### Motivation and Context

Fixes #6442 

### How Has This Been Tested?

- in browser
- via new tests for the bug

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by tim-lai** [comment]
@mathis-m looks like linting errors that fail CI

**Comment by mathis-m** [comment]
@tim-lai gonna fix that in some hours

**Comment by mathis-m** [comment]
My approach is to override the response schema's example in case there is a matching media type example defined.
And use the new schema with the getSampleSchema util to generate a matching example.

**Review by Rmkasun — APPROVED** [review]

File: src/core/components/response.jsx [hunk 1]
```diff
@@ -118,38 +117,60 @@ export default class Response extends React.Component {
...
     }
 
+    const overrideSchemaExample = (oldSchema, newExample) => {
+      if(newExample === undefined)
+        return oldSchema
+
+      if(!oldSchema)
+        oldSchema = { }
+
+      oldSchema.example = typeof newExample.toJS === "function"
```
Comment (tim-lai) [inline src/core/components/response.jsx:127]: let's use `isFunc` from `core/utils` here.

**Review by tim-lai — CHANGES_REQUESTED** [review]

**Comment by tim-lai** [comment]
still failing CI. this time in `bugs/5458.js`

File: src/core/components/response.jsx (hunk 1 above)
Comment (mathis-m) [reply src/core/components/response.jsx]: fixed with 3db8f62

**Comment by mathis-m** [comment]
> still failing CI. this time in `bugs/5458.js`

fixed.

**Comment by mathis-m** [comment]
> My approach is to override the response schema's example in case there is a matching media type example defined.
> And use the new schema with the getSampleSchema util to generate a matching example.

In case of xml this will not work as described in #6470

**Review by tim-lai — APPROVED** [review]

**Pull request merged**